crates/editor/src/display_map.rs

```diff
@@ -983,8 +983,23 @@ impl DisplaySnapshot {
                     break;
                 }
             }
-            let end = end.unwrap_or(max_point);
-            Some((start..end, self.fold_placeholder.clone()))
+            let mut adjusted_end = end.unwrap_or(max_point);
+
+            // Adjust end point to exclude folding any line breaks if present
+            while adjusted_end.row > 0
+                && self
+                    .buffer_snapshot
+                    .is_line_blank(MultiBufferRow(adjusted_end.row))
+            {
+                // Keep searching upwards until we find a non-blank line
```

Comment:
One concern I have is whether `start..adjusted_end` is legal, when we go back past `start` (as, in theory, this code can do?)

We're lucky that Rust is fine with such ranges and considers them empty, but I think it's better to return `None` or, if there's a meaning in the `adjusted_end..start` range, use that?

Reply:
I have changed the condition to stop the loop if the row goes past the start. Instead of a `> 0` check, it now checks `> start.row`. I believe this should resolve the issue, but we could also consider returning None or adjusting the range if needed. Let me know your thoughts!